### DISCUSSION TITLE
Java Agent Gauge API only works for the first value

### DISCUSSION THREAD

**Discussion by willardwang** [apache/skywalking#10857] [Q&A] [answered]
When I was using Gauge to measure the number of online users, the Gauge API confused me. When I built Guage for the same metric multiple times, only the first value took effect. As shown in the figure, putIfAbsent in the code is only valid for the first put, which makes subsequent put meter invalid, and there is no mechanism to clear meterMap (except for BootService shutdown）, I suspect I used it in a wrong way, looking forward to your guidance, thank you;
![image](https://github.com/apache/skywalking/assets/3304388/2a78ce0f-de17-43da-90ed-18df4b7be2fe)
![image](https://github.com/apache/skywalking/assets/3304388/6cfb4c3b-fde3-4058-912a-206b0ff54c7d)

**Comment by wu-sheng** [comment]
That is for meter creation. Why do you want to create it more than once? It is wrong from our perspective.

**Comment by wu-sheng** [comment, marked as answer]
Follow the doc, https://skywalking.apache.org/docs/skywalking-java/next/en/setup/service-agent/java-agent/java-plugin-development-guide/#meter-plugin

This is how to use this meter API.

**Reply by willardwang** [reply]
Thank you for your reply. Taking the number of online users as an example, I think the principle is as follows: when the instantaneous number of online users is obtained from the appropriate method interceptor and reported to the OAP, multiple values will be obtained at different times, and multiple values will form a set of time series curves. For the agent, I thought the meterMap would be clear after sending OAP

**Reply by willardwang** [reply]
Thank you. I have read this document for many times. Please forgive me for my misunderstanding

**Reply by wu-sheng** [reply]
Keep the created metrics, and call `increase`/`set`. Don't create every time, and the counter/guage is going to be sent.

**Reply by willardwang** [reply]
Thank you, I understand the clever design, so that the client does not care when the trigger is triggered (intercept or timing), but the agent will retrieve the agreed object every 20 seconds
